### PR TITLE
Add types declaration file

### LINKED ISSUE #1: TypeScript type declaration file
If you want I can write a TypeScript type declaration file for this package so that TS developers can utilise the type-checking feature of the language as they use this package. Let me know if you are interested.

### PR DESCRIPTION
**Pull request by geopic**
Resolves #1. Let me know if there are any issues.

### CODE REVIEW

File: politely.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export default function politely(config: {
+  services: {
+    start(): Promise<void>;
+    stop(): Promise<void>;
+  }[],
+  timeout: number;
+  logger: object;
```
Comment (raymond-lam) [inline politely.d.ts:7]: Can we be more specific than `object` for `logger`?  `logger` should be an object with `info`, `warn`, and `error` methods. A Winston logger, for instance, would satisfy this requirement. It is also an optional argument.

File: politely.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export default function politely(config: {
+  services: {
+    start(): Promise<void>;
+    stop(): Promise<void>;
+  }[],
+  timeout: number;
```
Comment (raymond-lam) [inline politely.d.ts:6]: timeout is an optional argument

**Review by raymond-lam — CHANGES_REQUESTED** [review]

File: politely.d.ts (hunk 2 above)
Comment (geopic) [reply politely.d.ts:6]: Noted

File: politely.d.ts (hunk 1 above)
Comment (geopic) [reply politely.d.ts]: Noted :+1:

File: politely.d.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export default function politely(config: {
+  services: {
+    start(): Promise<void>;
+    stop(): Promise<void>;
+  }[],
+  timeout?: number;
+  logger?: {
+    error(message: string): void;
```
Comment (raymond-lam) [inline politely.d.ts:8]: Although we only ever call `error`, `info` and `warn` with a single argument, a `winston` logger can accept a variable number of arguments. Will this be an issue in terms of static type checking?

**Review by raymond-lam — CHANGES_REQUESTED** [review]

File: politely.d.ts (hunk 3 above)
Comment (raymond-lam) [reply politely.d.ts:8]: (For that matter, I assume it will be okay if the logger that is passed in has these methods and more, right?)

Comment (raymond-lam) [reply politely.d.ts:8]: I just tried it and it worked

**Review by raymond-lam — APPROVED** [review]

**Pull request merged**